src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -120,5 +148,145 @@ public void ReadEndByteString()
+            {
+                source.Slice(o, l).CopyTo(destination);
+                destination = destination.Slice(l);
+            }
+
+            bytesWritten = concatenatedBufferSize;
+            AdvanceBuffer(offset);
+            DecrementRemainingItemCount();
+            return true;
+        }
+
+        private bool TryReadChunkedTextStringConcatenated(Span<char> destination, out int charsWritten)
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.TextString, out int offset, out int _);
+            ReadOnlySpan<byte> buffer = _buffer.Span;
+
+            int concatenatedStringSize = 0;
+            foreach ((int o, int l) in ranges)
+            {
+                concatenatedStringSize += s_utf8Encoding.GetCharCount(buffer.Slice(o, l));
+            }
+
+            if (concatenatedStringSize > destination.Length)
+            {
+                charsWritten = 0;
+                return false;
+            }
+
+            foreach ((int o, int l) in ranges)
+            {
+                s_utf8Encoding.GetChars(buffer.Slice(o, l), destination);
+                destination = destination.Slice(l);
+            }
+
+            charsWritten = concatenatedStringSize;
+            AdvanceBuffer(offset);
+            DecrementRemainingItemCount();
+            return true;
+        }
+
+        private byte[] ReadChunkedByteStringConcatenated()
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.ByteString, out int offset, out int concatenatedBufferSize);
+            var output = new byte[concatenatedBufferSize];
+
+            ReadOnlySpan<byte> source = _buffer.Span;
+            Span<byte> target = output;
+
+            foreach ((int o, int l) in ranges)
+            {
+                source.Slice(o, l).CopyTo(target);
+                target = target.Slice(l);
+            }
+
+            Debug.Assert(target.IsEmpty);
+            AdvanceBuffer(offset);
+            DecrementRemainingItemCount();
+            return output;
+        }
+
+        private string ReadChunkedTextStringConcatenated()
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.TextString, out int offset, out int concatenatedBufferSize);
+            ReadOnlySpan<byte> buffer = _buffer.Span;
+            int concatenatedStringSize = 0;
+
+            foreach ((int o, int l) in ranges)
+            {
+                concatenatedStringSize += s_utf8Encoding.GetCharCount(buffer.Slice(o, l));
+            }
+
+            var output = new char[concatenatedStringSize];
+            Span<char> target = output;
+
+            foreach ((int o, int l) in ranges)
+            {
+                s_utf8Encoding.GetChars(buffer.Slice(o, l), target);
+                target = target.Slice(l);
+            }
+
+            Debug.Assert(target.IsEmpty);
+            AdvanceBuffer(offset);
```

Comment:
To avoid confusion with the list tuple type field and the loop locals; I'd recommend using the name of the parameter from the helper method: `encodingLength`

Reply:
Thanks, was intending to rename but forgot.